CorsixTH/Src/th_gfx.cpp

```diff
@@ -242,22 +281,21 @@ bool animation_manager::load_from_th_file(
   // Read elements.
   size_t iSpriteCount = sheet->get_sprite_count();
   for (size_t i = 0; i < iElementCount; ++i) {
-    const th_element_properties* pTHElement =
-        reinterpret_cast<const th_element_properties*>(pElementData) + i;
+    const th_element_properties thElement(pElementData +
+                                          i * th_element_properties::size);
 
     element oElement;
-    oElement.sprite = pTHElement->table_position / 6;
-    oElement.flags = pTHElement->flags & 0xF;
-    oElement.x = static_cast<int>(pTHElement->offx) - 141;
-    oElement.y = static_cast<int>(pTHElement->offy) - 186;
-    oElement.layer = static_cast<uint8_t>(
-        pTHElement->flags >> 4);  // High nibble, layer of the element.
+    oElement.sprite = thElement.table_position / 6;
```

Comment:
`6` is actually `th_sprite_properties::size` ?

`// Size of a sprite.` works too of course.

Reply:
That predated my change and I didn't know what it was. That makes enough sense to change